test_util/integration_test.py

```diff
@@ -871,6 +871,87 @@ def test_if_we_have_capabilities(cluster):
     assert {'name': 'PACKAGE_MANAGEMENT'} in r.json()['capabilities']
 
 
+def test_octarine_http(cluster, timeout=30):
+    """
+    Test if we are able to send traffic through octarine.
+    """
+
+    test_uuid = uuid.uuid4().hex
+    proxy = ('"http://127.0.0.1:$(/opt/mesosphere/bin/octarine ' +
+             '--client --port marathon)"')
+    check_command = 'curl --fail --proxy {} marathon.mesos'.format(proxy)
+
+    app_definition = {
+        'id': '/integration-test-app-octarine-http-{}'.format(test_uuid),
+        'cpus': 0.1,
+        'mem': 128,
+        'ports': [10001],
+        'cmd': '/opt/mesosphere/bin/octarine marathon',
+        'disk': 0,
+        'instances': 1,
+        'healthChecks': [{
+            'protocol': 'COMMAND',
+            'command': {
+                'value': check_command
+            },
+            'gracePeriodSeconds': 5,
+            'intervalSeconds': 10,
+            'timeoutSeconds': 10,
+            'maxConsecutiveFailures': 3
+        }]
+    }
+
+    cluster.deploy_marathon_app(app_definition)
+
+
+def test_octarine_srv(cluster, timeout=30):
+    """
+    Test resolving SRV records through octarine.
+    """
+
+    # Limit string length so we don't go past the max SRV record length
+    test_uuid = uuid.uuid4().hex[16]
+    proxy = ('"http://127.0.0.1:$(/opt/mesosphere/bin/octarine ' +
+             '--client --port marathon)"')
+    port_name = 'pinger'
+    cmd = ('/opt/mesosphere/bin/octarine marathon & '
+           'chmod 755 linux-amd64 && ' +
+           './linux-amd64 -listener=:${PORT0} -say-string=version1')
+    raw_app_id = 'integration-test-app-octarine-srv-{}'.format(test_uuid)
+    check_command = ('curl --fail --proxy {} _{}._{}._tcp.marathon.mesos')
+    check_command = check_command.format(proxy, port_name, raw_app_id)
+
+    app_definition = {
+        'id': '/{}'.format(raw_app_id),
+        'cpus': 0.1,
+        'mem': 128,
+        'cmd': cmd,
+        'disk': 0,
+        'instances': 1,
+        'portDefinitions': [
+          {
+            'port': 10002,
+            'protocol': 'tcp',
+            'name': port_name,
+            'labels': {}
+          }
+        ],
+        'uris': ['https://s3.amazonaws.com/sargun-mesosphere/linux-amd64'],
```

Comment:
Can we get this to a place we control better?

Ideally: a build job in CI that rebuilds and pushes it if it changes. It's fine having this inside this repo too.

That way we are sure the whole tooling is always together.